_inc/search/components/search-widget.jsx

```diff
@@ -0,0 +1,78 @@
+/** @jsx h */
+
+/**
+ * External dependencies
+ */
+import { h, Component } from 'preact';
+import Portal from 'preact-portal';
+import { debounce } from 'lodash';
+
+class SearchWidget extends Component {
+	constructor() {
+		super( ...arguments );
+		this.requestId = 0;
+		this.state = {
+			query: this.props.initialValue,
+			results: [],
+		};
+		this.onChangeQuery = this.onChangeQuery.bind( this );
+		this.getResults = debounce( this.getResults.bind( this ), 500 );
+		this.getResults( this.props.initialValue );
+	}
+	componentDidMount() {
+		if ( this.props.grabFocus ) {
+			this.input.focus();
+		}
+	}
+
+	bindInput = input => ( this.input = input );
+	onChangeQuery = event => {
+		const query = event.target.value;
+		this.setState( { query } );
+		this.getResults( query );
+	};
+
+	getResults( query ) {
+		if ( query ) {
+			const { api } = this.props;
+			if ( api ) {
+				this.requestId++;
+				const requestId = this.requestId;
+				api
+					.fetch( query )
+					.then( response => {
+						return response.json();
+					} )
+					.then( json => {
+						if ( this.requestId === requestId ) {
+							this.setState( { results: json } );
+						}
+					} );
+			}
+		} else {
+			this.setState( { results: [] } );
+		}
+	}
+
+	render() {
+		const { query, results } = this.state;
+		const { SearchResults, api } = this.props;
+		return (
+			<div>
+				<p>
+					<input
+						type="text"
+						value={ query }
+						onInput={ this.onChangeQuery }
+						ref={ this.bindInput }
+					/>
+				</p>
+				<Portal into="#results">
```

Comment:
What would happen with multiple search widgets on one page? Should you include a running unique integer?

Reply:
These ids should be props passed to the widget, definitely.

Reply:
Added to #13027